### ISSUE TITLE
Shadow support for SVG's

### ISSUE THREAD

**Issue by wafisher** [mapnik/mapnik#4025]
Would it be possible to get support for drop shadows in SVG's? I'm using Mapbox (which relies on Mapnik) with a custom SVG but neither `filter` nor `blur` is supported in Mapnik per https://github.com/mapnik/mapnik/wiki/SVG-support, making it hard to include drop shadows in any way since those features get stripped away.

Would love to see support for drop shadows if possible!

**Comment by artemp** [comment]
@wafisher - it would be nice, agree. I'll might find inspiration to work on this feature so I'll keep this issue open, thanks.

**Comment by wafisher** [comment]
Hi, I don't suppose there's a good workaround for this in the meantime? Perhaps with supported properties?

**Comment by wafisher** [comment]
Hi, would still love to see support for this or a workaround, if possible. Perhaps updates over the last two years might hint at some workaround?

**Comment by artemp** [comment]
https://dev.w3.org/SVG/modules/filters/publish/SVGFilter.html - it's a handful :)

@wafisher - if you have access to Adobe Illustrator, could you try creating simple SVG with a shadow and exporting it as `SVG Tiny 1.2`?

**Comment by wafisher** [comment]
I was able to create the same SVG in Adobe Illustrator. I've uploaded it here. I'm using Mapbox to display these SVG's and the shadow didn't carry into Mapbox so I'm assuming it's not supported by Mapnik.

[DropletShadowTiny.svg.zip](https://github.com/mapnik/mapnik/files/6259309/DropletShadowTiny.svg.zip)

That is quite a bit. Perhaps you could focus on one or two of those filter primitives like `feDropShadow`?

**Comment by artemp** [comment]
> I was able to create the same SVG in Adobe Illustrator. I've uploaded it here. I'm using Mapbox to display these SVG's and the shadow didn't carry into Mapbox so I'm assuming it's not supported by Mapnik.

Yes, SVG rendered in mapnik doesn't support shadows created by using filters.

> 
> [DropletShadowTiny.svg.zip](https://github.com/mapnik/mapnik/files/6259309/DropletShadowTiny.svg.zip)

This is exactly what I was looking for, thanks.
> That is quite a bit. Perhaps you could focus on one or two of those filter primitives like `feDropShadow`?

In DropletShadowTiny.svg `shadow` was exported as `<image>` element, `feDropShadow` is not part of Tiny 1.2 profile.

Adding support for `<image>` elements would be quite straightforward but there are portability and security issues, see https://tools.ietf.org/id/draft-svg-tiny-ps-abrotman-00.txt 
On the other hand `filters` are a large area.. this is why it got stalled :) I'll have to ponder on your proposal to just add `feDropShadow` filter

**Comment by artemp** [comment]
@wafisher could you post original DropletShadow.svg (with feDropShadow filter), pls?

**Comment by wafisher** [comment]
Sure, here's the original. Sketch actually doesn't use `feDropShadow` when exporting; it uses a combination of other filters, which is another popular approach to creating a drop shadow. But I imagine it'd be easiest for you to implement `feDropShadow` and not the rest.

[DropletShadowSketch.svg.zip](https://github.com/mapnik/mapnik/files/6266410/DropletShadowSketch.svg.zip)

(Ignore the `Rectangle` component as that's just a placeholder for Mapbox.)

**Comment by wafisher** [comment]
Hi, would still love to see this. If it's any motivation at all, this feature would land in a (Mapbox-heavy) app that is used by 100K people/day.

Here's what our trains look like now without the right shadow. You can see the tip doesn't have a shadow under it

![IMG_2739](https://user-images.githubusercontent.com/3514570/153617173-7f34259b-3495-4fe8-9402-64c974508ea2.PNG)
.

**Comment by wafisher** [comment]
Hi, still interested in this and still working on the same product! Mostly posting for fun.
